src/graphql/historicalProposal.utils.ts

```diff
@@ -0,0 +1,39 @@
+import { FragmentType, getFragmentData } from 'gql'
+import { ProposalItemFragmentDoc } from 'gql/graphql'
+
+import { UIProposal, UIProposalMetadata } from 'types'
+import { ProposalStatus } from 'util/enums'
+
+export function nodeToUIProposal(node: FragmentType<typeof ProposalItemFragmentDoc>) {
+  const proposal = getFragmentData(ProposalItemFragmentDoc, node)
+  if (!proposal) return null
+  const executorResult =
+    {
+      PROPOSAL_EXECUTOR_RESULT_UNSPECIFIED: 0,
+      PROPOSAL_EXECUTOR_RESULT_NOT_RUN: 1,
+      PROPOSAL_EXECUTOR_RESULT_SUCCESS: 2,
+      PROPOSAL_EXECUTOR_RESULT_FAILURE: 3,
+      UNRECOGNIZED: -1,
+    }[proposal.executorResult] || -1
```

Comment:
nit: we could probably add this object to the others in `src/util/enums`

Reply:
485220e

ended up making a helper function, i think it's a bit more helpful than just putting an object there